plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/ProjectStatePersister.java

```diff
@@ -329,12 +376,14 @@ public PersistedState readProjectState(InputStream stream, String expectedLangua
 			throws IOException, ClassNotFoundException {
 
 		int version = stream.read();
-		if (version != CURRENT_VERSION) {
+		if (version != CURRENT_VERSION && version != VERSION_1) {
 			return null;
 		}
 
-		try (ObjectInputStream input = new ObjectInputStream(
-				new BufferedInputStream(new GZIPInputStream(stream, 8192)))) {
+		try (BufferedInputStream buffy = new BufferedInputStream(new GZIPInputStream(stream, 8192));
+				InputStream dataOrObjectInputStream = version == VERSION_1 ? new ObjectInputStream(buffy)
```

Comment:
I see. How about not having Version#1? At the moment, we don't have regular users yet.

Reply:
The logged exceptions were annoying when switching branches ;)

Apart from that, it would be ok to remove the backwards compat there. Regular users would see a rebuild in any case since the language version would be differnt.